File: 1217-minimum-cost-to-move-chips-to-the-same-position/1217-minimum-cost-to-move-chips-to-the-same-position.py
class Solution:
    def minCostToMoveChips(self, position: List[int]) -> int:
        length_evens = len([x for x in position if x % 2 == 0])
        length_odds = len(position) - length_evens
        
        return min(length_evens, length_odds)

File: 1217-minimum-cost-to-move-chips-to-the-same-position/1217-minimum-cost-to-move-chips-to-the-same-position.java
class Solution {
    public int minCostToMoveChips(int[] position) {
        int even = 0;
        for(int i:position)
            even += (i%2==0) ? 1 : 0; // get # of evens
        return even > position.length/2 ? position.length-even : even; 
    }
}